src/Widgets/DataFormNumberWidget/DataFormNumberWidgetComponent.tsx

```diff
@@ -56,18 +73,39 @@ provideComponent(DataFormNumberWidget, ({ widget }) => {
         </>
       ) : null}
       <br />
-      <input
-        className="form-control"
-        defaultValue={defaultValue}
-        id={id}
-        name={attributeName ?? ''}
-        placeholder={widget.get('placeholder')}
-        required={widget.get('required')}
-        type="number"
-        min={widget.get('minValue') ?? undefined}
-        max={widget.get('maxValue') ?? undefined}
-        step={widget.get('stepValue') ?? undefined}
-      />
+      <div className="input-group flex-nowrap">
+        <button
+          aria-label="-"
+          className="btn btn-primary btn-sm"
+          disabled={hasValue && min !== undefined && defaultValue <= min}
+          onClick={down}
+        >
+          <i className="bi bi-dash-lg px-0 text-white" />
+        </button>
+        <input
+          className="form-control text-center no-arrows"
+          defaultValue={defaultValue}
+          id={id}
+          max={max}
+          min={min}
```

Comment:
There is an issue with invalid inputs:

https://github.com/user-attachments/assets/31d29786-9ea6-4e26-a01c-8fe305d91b65


How to reproduce:
* Add something to the cart
* Adjust the quantity to a valid value
* Switch to the cart
* Try to input something invalid into the quantity field (e.g. `-1` or `some text`)
* Submit the form

_Expected_
I'm not 100% sure, but maybe the form should reject the submission before the value is corrected.

_Actual_
The old "valid" value of the number is used.

This is probably not a new issue. Could you still take a look on how this could be improved?

Reply:
Two issues.
1. I added a commit that reports the validity issues.
2. The cart checkout button is outside of the form(s) for the individual items. Not sure if and how we should/could tackle this.

Reply:
Idea for the `CheckoutButtonWidget`:
1. `DataFromContainerWidget` adds `data-class-name={dataItem?.dataClass().name()}` and `data-class-id={dataItem?.id()}` to the `<form>` tag (maybe with better naming).
2. `CheckoutButtonWidget` adds a check, that all CartItem forms are valid. E.g. `[...document.querySelectorAll(`form[data-class-name='CartItem']`)].every(f => f.reportValidity())`.

WDTY?

Reply:
Currently we don't know enough about the relevant parts of the page. The `data` approach may learn too much about potentially unrelated parts of the page.

Could we add a validation to the checkout button to make sure it's part of a form widget?

Reply:
> Could we add a validation to the checkout button to make sure it's part of a form widget?

No, because there is one form per cart item, but only one checkout button in total.

Reply:
This could be fixed in a follow up.

Reply:
How about: On checkout, we check if there is **any** invalid form on the page, optimistically assume the checkout page has only (invalid) forms related to checkout?

Reply:
Sounds good to me.